node/src/types/node_config.rs

```diff
@@ -3,11 +3,22 @@ use serde::{Deserialize, Serialize};
 
 use crate::types::BlockHash;
 
-/// Node configuration.
-#[derive(Default, DataSize, Debug, Deserialize, Serialize)]
+/// Node fast-sync configuration.
+#[derive(Default, DataSize, Debug, Deserialize, Serialize, Clone)]
 // Disallow unknown fields to ensure config files and command-line overrides contain valid keys.
 #[serde(deny_unknown_fields)]
 pub struct NodeConfig {
-    /// Hash used as a trust anchor when joining, if any.
+    /// Hash used as a trust anchor when joining. If no hash is provided, attempt to run genesis.
     pub trusted_hash: Option<BlockHash>,
+
+    /// Whether to run in archival-sync mode. Archival-sync mode captures all data (blocks, deploys
+    /// and global state) back to genesis. Defaults to `false`.
+    #[serde(default = "NodeConfig::default_archival_sync_status")]
```

Comment:
I know we have a mixture of optional and non-optional fields in our config.toml files, but I think the vast majority are non-optional, so we should generally follow that approach.

One advantage is that we can't forget to add the field to the config.toml using that approach.

Reply:
I made this no longer optional in ba89b36